libvmi/driver/kvm/kvm.c

```diff
@@ -1364,6 +1380,17 @@ kvm_init_vmi(
     }
     vmi->num_vcpus = info.nrVirtCpu;
 
+#ifdef HAVE_LIBVMI_REQUEST
```

Comment:
The whole point of checking the QEMU version is to determine if not having the request structure defined by a header is a problem or not. So right now you have it backwards. If the pre-processor check HAVE_LIBVMI_REQUEST succeeds we know we are 2.8+ and have the libvmi header. What we need is to check the runtime version and if it is 2.8+ and we don't have HAVE_LIBVMI_REQUEST then print an error message and exit.